glass/src/lib/native/include/glass/support/ExpressionParser.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <stdint.h>
+
+namespace glass::expression {
+
+enum class ParseResultKind {
+  Success,
+  Error,
+};
+
+template <typename V>
+struct ParseResult {
+  static ParseResult<V> CreateSuccess(V value);
+  static ParseResult<V> CreateError(const char* errorMessage);
+
+  // If kind is Success, expression value is in successVal
+  // If kind is Error, error message is in errorMessage
+  ParseResultKind kind;
+  union {
+    V successVal;
+    const char* errorMessage;
+  };
```

Comment:
Use `wpi::expected` instead, in `<wpi/expected>` (it's a shim for `std::expected`, which isn't available until C++23).

Reply:
It appears I am unable to include <wpi/expected> since wpi::detail::conjuction is defined in both `wpiutil/src/main/native/thirdparty/expected/include/wpi/expected` and `wpiutil/src/main/native/thirdparty/json/include/wpi/detail/meta/type_traits.h`. If I include expected from `ExpressionParser.h`, `NetworkTables.cpp` includes both. I'm not really sure how to resolve this since both are needed in `NetworkTables.cpp`. 

Full compiler error message:

<details>
<summary>How do I dropdown?</summary>
```
In file included from /home/ultraviolet/github/allwpilib/glass/src/lib/native/include/glass/support/ExpressionParser.h:10,
                 from /home/ultraviolet/github/allwpilib/glass/src/libnt/native/cpp/NetworkTables.cpp:36:
/home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/expected/include/wpi/expected:238:28: error: redefinition of ‘struct wpi::detail::conjunction< <template-parameter-1-1> >’
  238 | template <class...> struct conjunction : std::true_type {};
      |                            ^~~~~~~~~~~
In file included from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/detail/exceptions.h:22,
                 from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/detail/conversions/from_json.h:23,
                 from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/adl_serializer.h:14,
                 from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/json.h:35,
                 from /home/ultraviolet/github/allwpilib/glass/src/libnt/native/include/glass/networktables/NetworkTables.h:20,
                 from /home/ultraviolet/github/allwpilib/glass/src/libnt/native/cpp/NetworkTables.cpp:5:
/home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/detail/meta/type_traits.h:189:27: note: previous definition of ‘struct wpi::detail::conjunction< <template-parameter-1-1> >’
  189 | template<class...> struct conjunction : std::true_type { };
      |                           ^~~~~~~~~~~
In file included from /home/ultraviolet/github/allwpilib/glass/src/lib/native/include/glass/support/ExpressionParser.h:10,
                 from /home/ultraviolet/github/allwpilib/glass/src/libnt/native/cpp/NetworkTables.cpp:36:
/home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/expected/include/wpi/expected:239:27: error: redefinition of ‘struct wpi::detail::conjunction<B>’
  239 | template <class B> struct conjunction<B> : B {};
      |                           ^~~~~~~~~~~~~~
In file included from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/detail/exceptions.h:22,
                 from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/detail/conversions/from_json.h:23,
                 from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/adl_serializer.h:14,
                 from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/json.h:35,
                 from /home/ultraviolet/github/allwpilib/glass/src/libnt/native/include/glass/networktables/NetworkTables.h:20,
                 from /home/ultraviolet/github/allwpilib/glass/src/libnt/native/cpp/NetworkTables.cpp:5:
/home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/detail/meta/type_traits.h:190:26: note: previous definition of ‘struct wpi::detail::conjunction<B>’
  190 | template<class B> struct conjunction<B> : B { };
      |                          ^~~~~~~~~~~~~~
In file included from /home/ultraviolet/github/allwpilib/glass/src/lib/native/include/glass/support/ExpressionParser.h:10,
                 from /home/ultraviolet/github/allwpilib/glass/src/libnt/native/cpp/NetworkTables.cpp:36:
/home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/expected/include/wpi/expected:241:8: error: redefinition of ‘struct wpi::detail::conjunction<B, Bn ...>’
  241 | struct conjunction<B, Bs...>
      |        ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/detail/exceptions.h:22,
                 from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/detail/conversions/from_json.h:23,
                 from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/adl_serializer.h:14,
                 from /home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/json.h:35,
                 from /home/ultraviolet/github/allwpilib/glass/src/libnt/native/include/glass/networktables/NetworkTables.h:20,
                 from /home/ultraviolet/github/allwpilib/glass/src/libnt/native/cpp/NetworkTables.cpp:5:
/home/ultraviolet/github/allwpilib/wpiutil/src/main/native/thirdparty/json/include/wpi/detail/meta/type_traits.h:192:8: note: previous definition of ‘struct wpi::detail::conjunction<B, Bn ...>’
  192 | struct conjunction<B, Bn...>
      |        ^~~~~~~~~~~~~~~~~~~~~
```

Reply:
Fixed it, turns out `NetworkTables.cpp` was including `ExpressionParser.h` when it did not need to.